_posts/2018/2018-05-27-first-and-last-may-not-mean-what-you-think.md

```diff
@@ -0,0 +1,58 @@
+---
+title: Active Record’s first and last may not mean what you think
+description: 'Naming truly is an art'
+layout: article
+category: ruby
+image:
+  base: '2018/first-and-last-may-not-mean-what-you-think'
+  alt: 'Running track'
+  source: 'https://unsplash.com/photos/52p1K0d0euM'
+  credit: 'Austris Augusts'
+---
+
+Each Active Record model comes with the well-used ‘get me a single record’ scopes: `.first` and `.last`.
+
+## Instead of…
+
+...using the methods directly on the base ActiveRecord class.
+
+```ruby
+User.first
+User.last
+```
+
+
+## Use…
+
+...them only with ordered scopes. Ideally with named concepts.
+
+### `app/models/user.rb`
+
+```ruby
+class User < ActiveRecord::Base
+  scope :by_created, -> { order(created_at: :asc) }
+  scope :earliest_created, -> { by_created.first }
+  scope :most_recently_created, -> { by_created.last }
+  # ...
+end
+
+User.most_recently_created
+```
+
+
+## But why?
+
+This is mostly about making the code more explicit.
+
+The `.first` and `.last` methods, when called on a bare Active Record model scope, are colloquially used to mean ‘created first’ and ‘created most recently’.
+
+It’s only an accident of Rails default design that sorting by ascending `id` is the same order as the ascending `created_at` timestamp.
```

Comment:
*Rails'

Reply:
What you're actually trying to say here is that:  first returns the smallest number ID, and last returns the largest number ID. As you create objects, Rails assigns IDs incrementally. Therefore, it's only by luck that when you call 'first' you get the 'earliest created' — really you're asking Rails for: give me the lowest number ID.

If that's correct, I think that there must be a way to take what I've said above and improve your paragraph — not quite sure what that improvement is yet.

Maybe even merge in the paragraph before where you refer to the 'colloquial' use of 'first' and 'last'.